src/libecalc/presentation/exporter/infrastructure.py

```diff
@@ -0,0 +1,405 @@
+from datetime import datetime
+from typing import Iterable, List, Optional, Tuple
+
+from typing_extensions import assert_never
+
+from libecalc.application.graph_result import GraphResult
+from libecalc.common.component_type import ComponentType
+from libecalc.common.errors.exceptions import ProgrammingError
+from libecalc.common.logger import logger
+from libecalc.common.temporal_model import TemporalModel
+from libecalc.common.time_utils import Frequency, Period, Periods
+from libecalc.common.units import Unit
+from libecalc.common.utils.rates import TimeSeries, TimeSeriesFloat, TimeSeriesRate, TimeSeriesStreamDayRate
+from libecalc.core.result import GeneratorSetResult
+from libecalc.dto import FuelConsumer, GeneratorSet
+from libecalc.dto.utils.validators import convert_expression
+from libecalc.presentation.exporter.domain.exportable import (
+    Attribute,
+    AttributeMeta,
+    AttributeSet,
+    Exportable,
+    ExportableSet,
+    ExportableType,
+)
+from libecalc.presentation.yaml.yaml_types.emitters.yaml_venting_emitter import (
+    YamlDirectTypeEmitter,
+    YamlOilTypeEmitter,
+    YamlVentingType,
+)
+
+
+class TimeSeriesAttribute(Attribute):
+    def __init__(self, time_series: TimeSeries, attribute_meta: AttributeMeta):
+        self._attribute_meta = attribute_meta
+        self._time_series = time_series
+
+    def datapoints(self) -> Iterable[Tuple[datetime, float]]:
+        return self._time_series.datapoints()
+
+    def get_meta(self) -> AttributeMeta:
+        return self._attribute_meta
+
+
+class InstallationExportable(Exportable):
+    def __init__(self, installation_graph: GraphResult):
+        self._installation_graph = installation_graph
+        self._frequency: Optional[Frequency] = None
+        self._installation_dto = installation_graph.graph.get_node(installation_graph.graph.root)
+
+    def get_electricity_production(self, unit: Unit) -> AttributeSet:
+        attributes = []
+        for fuel_consumer in self._installation_dto.fuel_consumers:
+            if not isinstance(fuel_consumer, GeneratorSet):
+                continue
+
+            fuel_consumer_result = self._installation_graph.get_energy_result(fuel_consumer.id)
+            assert isinstance(fuel_consumer_result, GeneratorSetResult)
+            consumer_category = TemporalModel(fuel_consumer.user_defined_category)
+            for period, category in consumer_category.items():
```

Comment:
```suggestion
            for period, category in consumer_category.items():
            
                if category == "POWER-FROM-SHORE" and fuel_consumer.cable_loss is None:
                    # Edge case; we don't want to assume cable loss = 0 if power from shore. So we skip reporting produced electricity if cable loss isn't specified
                    continue
```

Adding a check here would solve the extra power supply column, but that is something we want to avoid IMO

Reply:
I agree that we should not handle edge cases like that, it will never end ...